frontend/src/utilities/__tests__/useDraggableTable.spec.ts

```diff
@@ -0,0 +1,322 @@
+import { act } from '@testing-library/react';
+import useDraggableTable from '~/utilities/useDraggableTable';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+
+let setItemOrder = jest.fn();
+beforeEach(() => {
+  setItemOrder = jest.fn();
+});
+
+afterEach(() => {
+  setItemOrder.mockReset();
+});
+
+const generateRowsWithItems = (itemOrder: string[]) => {
+  const tbody = document.createElement('tbody');
+
+  itemOrder.forEach((itemId) => {
+    const row = document.createElement('tr');
+    row.id = itemId;
+    tbody.appendChild(row);
+  });
+
+  return tbody;
+};
+describe('initializeState', () => {
+  it('should initialize state and return tableProps and rowProps', async () => {
+    const renderResult = testHook(useDraggableTable)(['item1', 'item2', 'item3'], setItemOrder);
+    expect(renderResult).hookToHaveUpdateCount(1);
+    renderResult.rerender(['item1', 'item2', 'item3'], setItemOrder);
+
+    expect(renderResult).hookToHaveUpdateCount(2);
+    expect(renderResult.result.current.tableProps.tbodyProps.ref.current).toBe(null);
+    expect(renderResult.result.current.tableProps.className).toBe(undefined);
```

Comment:
You should have at least one assertion that ensures all callback properties are functions.
Have this test use `hookStrictEquals`.

Move this assertion before the `rerender`. The only reason for calling `rerender` is to assert stability here.